lib/portage/sync/modules/git/git.py

```diff
@@ -228,13 +228,16 @@ def update(self):
         if not self.verify_head(revision="refs/remotes/%s" % remote_branch):
             return (1, False)
 
-        if shallow:
+        merge_cmd = [
```

Comment:
Maybe we could conditionally do this based on whether the state is currently clean or not.